test/api/oran_o2ims_suite_test.go

```diff
@@ -0,0 +1,327 @@
+package api
+
+import (
+	"bytes"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestAPI(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "OranO2ims Test Suite")
+}
+
+var _ = BeforeSuite(func() {
+	Expect(os.Getenv("TEST_HOST")).NotTo(BeZero(), "Please make sure TEST_HOST is set correctly.")
+	if os.Getenv("TEST_HOST") == "" {
+		Skip("API tests were skipped because environment variable 'TEST_HOST' isn't set")
+	}
+})
+
+var _ = Describe("Metadata Server API testing", func() {
+	Context("When getting infrastructure Inventory API version", func() {
+		It("should return OK in the response and json response should match reference json", func() {
+			requestBody := []byte(`{"uriPrefix":"/o2ims-infrastructureInventory/v1","apiVersions":[{"version":"1.0.0"}]}`)
+			request, _ := http.NewRequest("GET", "https://"+os.Getenv("TEST_HOST")+"/o2ims-infrastructureInventory/api_versions", bytes.NewBuffer([]byte(requestBody)))
+			request.Header.Set("Content-Type", "application/json")
+
+			By("Executing https petition")
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			client := &http.Client{Transport: tr}
+			response, err := client.Do(request)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking OK status response")
+			Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+			By("Checking response JSON is equal to expected JSON")
+			responseBody, _ := io.ReadAll(response.Body)
+			Expect(responseBody).To(MatchJSON(requestBody))
+		})
+	})
+	Context("When getting infrastructure Inventory description", func() {
+		It("should return OK in the response and json response should match reference json", func() {
+			requestBody := []byte(`{"serviceUri":"https://` + os.Getenv("TEST_HOST") + `","extensions":{},"oCloudId":"f7fd171f-57b5-4a17-b176-9a73bf6064a4","globalCloudId":"f7fd171f-57b5-4a17-b176-9a73bf6064a4","name":"OpenShift O-Cloud","description":"OpenShift O-Cloud"}`)
+			request, _ := http.NewRequest("GET", "https://"+os.Getenv("TEST_HOST")+"/o2ims-infrastructureInventory/v1", bytes.NewBuffer([]byte(requestBody)))
+			request.Header.Set("Content-Type", "application/json")
+
+			By("Executing https petition")
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			client := &http.Client{Transport: tr}
+			response, err := client.Do(request)
+			Expect(err).NotTo(HaveOccurred())
+			By("Checking OK status response")
+			Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+			By("Checking response JSON is equal to expected JSON")
+			responseBody, _ := io.ReadAll(response.Body)
+			Expect(responseBody).To(MatchJSON(requestBody))
```

Comment:
Checking complete JSON matches is fragile. For example, if a new version of the software adds a new field then this will fail. An alternative to that complete JSON matching can be to check for the presence and values of specific fields. For example, you can use our `MatchJQ` custom matcher to extract a field (using a jq expression) and check its value:

```go
Expect(responseBody).To(MatchJQ(`.name`, "OpenShift O-Cloud"))
```

You can have multiple likes like that ^ to check multiple fields. It may be material for a separate pull request, but please consider it.